contributors/design-proposals/propagation.md

```diff
@@ -150,6 +165,42 @@ non-privileged', an environment check and an WARNING log will be emitted about
 whether propagation mode is supported.
 
 
+## Extra Concerns
+
+@lucab and @euank has some extra concerns about pod isolation when propagation
+modes are changed, listed below:
+
+1. how to clean such pod resources (as mounts are now crossing pod boundaries,
+thus they can be kept busy indefinitely by processes outside of the pod)
+
+1. side-effects on restarts (possibly piling up layers of full-propagation mounts)
+
+1. how does this interacts with other mount features (nested volumeMounts may or
+may not propagate back to the host, depending of ordering of mount operations)
+
+1. limitations this imposes on runtimes (RO-remounting may now affects the host,
+is it on purpose or a dangerous side-effect?)
+
+1. interaction with pod semantics (docker doesn't have a pod concept, but pod-
+aware runtimes may perform additional moves/remounts while preparing a pod)
+
+1. shared/slave flags refer to peer groups. If you want your mounts to be tied to the
+lifetime of the node, they must be performed as shared with the peer group of
+host's / (ie. usually 1). This may not always be possible under pod semantics.
+
+1. MS_MOVE is not allowed in shared subtree, which may make runtime implementation
+more complex and some scenario unfeasible (for example, creating a RO bind mount
+isn't an atomic operation and thus it is often mimicked as bind + remount-ro +
+move to appear in the pod context as RO from the start)
+- For the race-window issue, AFAIK, docker make mounts before user process (or
+ say the container) is running, and docker runs different container when k8s
+ ask it to restart, so this race-window should not harm.
+
+These concerns are valid and we decide to limit the propagation mode to HostPath
```

Comment:
This seems a new use case, do you mind adding some words in the "use case" section?